modules/apps/chargen/char-gen.js

```diff
@@ -340,19 +345,17 @@ export default class CharGenWfrp4e extends FormApplication {
 
   }
 
-  addStage(stage, index)
-  {
-    let stageObj = stage.stageData()
-    if (index == undefined)
-    {
+  addStage(stage, index, stageData = {}) {
+    let stageObj = stage.stageData();
+    stageObj = foundry.utils.mergeObject(stageObj, stageData);
+
+    if (index === undefined) {
```

Comment:
Would it be worth simply moving all stageData into the stage itself so this isn't needed?

Reply:
The Stage itself has its data hardcoded in the `static stageData()` method, the potential issue is that it's the same, always.

The main reason for that change is to allow to reuse the same Stage Class with different data (for example different `name` etc) and also to provide more data to the Stage. 

And since to add new stage we don't initialize the stage, but only pass a class, I don't know how else easily do this.

Unless we would switch to pass an already instantiated stage object instead of a class to the `addStage()`?

I'm curious what you have in mind

Reply:
> reuse the same Stage Class with different data (for example different name etc) 

I'm not sure I follow this, when would this ever be needed? Do you have an example? 

> And since to add new stage we don't initialize the stage, but only pass a class, I don't know how else easily do this.

Well, `stageData` is static, we wouldn't need to have an instance, unless I'm misunderstanding something. 

I'm not really leaning either way I just wonder if it'd be simpler to move the stageData into the class as static data, though I don't know what we would do with the `app` property.

Reply:
The only thing I actually have in mind is making a stage that allows to select items to start with. It could be used to pickup petty spells, arcane spells, custom fanmade spells, maybe some blessings or something.

I thought of adding something like that in my Armoury, mainly to pick the Runebound Spells, but then I realized that once I do it, then the exact same class could be used to allow players selecting their starting spells and maybe other things. All that would need to change is the title and maybe a callback or a flag that would control the possible selection. 

But then again, I guess I could just do one class and then extend it for every use case with absolute minimal required overrides... 

I didn't really think it super through, as I don't have immediate need for this, but what I commited seemed like the easiest thing to achieve that with as little change as possible. 

We can put it on hold for now if you prefer to sit on it

Reply:
Well I think that would simply be a case of subclassing some `SpellSelectionStage` and redefining its data.

Regardless, I don't think it's in the scope of this PR and can be changed later if needed.